nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java

```diff
@@ -0,0 +1,166 @@
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class DremioHelper {
+  private String token;
+  private String projectId;
+  private String baseUrl;
+  private String catalogName;
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private static JsonNode parseJson(String json) throws IOException {
+    JsonFactory factory = OBJECT_MAPPER.getFactory();
+    JsonParser parser = factory.createParser(json);
+    return OBJECT_MAPPER.readTree(parser);
+  }
+
+  DremioHelper(String projectId, String token, String baseUrl, String catalogName) {
+    this.projectId = projectId;
+    this.token = token;
+    this.baseUrl = baseUrl;
+    this.catalogName = catalogName;
+  }
+
+  private String createPayload(String query) {
+    String payload =
+        format("{ \"sql\": \"%s\", \"context\": [\"%s\", \"db\"] }", query, catalogName);
+    return payload;
+  }
+
+  private String readResponse(HttpURLConnection con, String url) throws IOException {
+    int responseCode = con.getResponseCode();
+    if (responseCode != HttpURLConnection.HTTP_OK) { // success
+      throw new IOException(
+          "Request for " + url + " was not successful with http code: " + responseCode);
+    }
+    try (BufferedReader in =
+        new BufferedReader(new InputStreamReader(con.getInputStream(), StandardCharsets.UTF_8))) {
+      String inputLine;
+      StringBuilder response = new StringBuilder();
+      while ((inputLine = in.readLine()) != null) {
+        response.append(inputLine);
+      }
+      return response.toString();
+    }
+  }
+
+  private String performHttpRequest(String url, String payload) throws IOException {
+    URL obj = new URL(url);
+    HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+    con.setRequestProperty("Authorization", "Bearer " + token);
+    con.setRequestProperty("Content-Type", "application/json");
+    if (payload != null) {
+      con.setRequestMethod("POST");
+      con.setDoOutput(true);
+      try (OutputStream os = con.getOutputStream()) {
+        os.write(payload.getBytes(StandardCharsets.UTF_8));
+      }
+    } else {
+      con.setRequestMethod("GET");
+    }
+    return readResponse(con, url);
+  }
+
+  private String waitForJobStatus(String jobId) throws IOException {
+    // The doc for getting the job status for cloud is not there, but it is similar to software
+    // See docs: https://docs.dremio.com/software/rest-api/jobs/get-job/
+    String jobState = "RUNNING";
+    String url = baseUrl + "/v0/projects/" + projectId + "/job/" + jobId;
+    Set<String> finalJobStatesList = new HashSet<>(asList("COMPLETED", "FAILED", "CANCELED"));
+    while (!finalJobStatesList.contains(jobState)) {
```

Comment:
Hm. This can become an endless loop, and it's querying for the job status "as fast as possible".

Should have a timeout here, and sleep between iterations to not put too much pressure on the backend.

Reply:
done.